main.go

```diff
@@ -231,10 +235,43 @@ func paConnectionWatchdog(ctx *ntcontext) {
 		ctx.paClient = paClient
 		go updateNoiseSupressorLoaded(paClient, &ctx.noiseSupressorState)
 
-		ctx.inputList = getSources(paClient)
+		ctx.inputList = getSourcesWithPreSelectedInput(paClient, ctx.config)
 
 		resetUI(ctx)
 
 		time.Sleep(500 * time.Millisecond)
 	}
 }
+
+func getSourcesWithPreSelectedInput(client *pulseaudio.Client, config *config) []input {
+	inputs := getSources(client)
+	preselectedInputId := config.LastUsedInput
+	inputExists := false
+	for _, input := range inputs {
+		inputExists = inputExists || input.ID == *preselectedInputId
+	}
+	if !inputExists {
+		defaultSource, err := getDefaultSourceID(client)
+		if err != nil {
+			log.Printf("Failed to load default source\n")
```

Comment:
Please also log the actual error

    log.Printf("Failed to load default source: %+v\n", err)